lib/active_model_serializers/model.rb

```diff
@@ -6,7 +6,21 @@ class Model
     include ActiveModel::Serializers::JSON
     include ActiveModel::Model
 
-    class_attribute :attribute_names
+    # Configuration to avoid a breaking change with older versions of this class which lacked defined attributes.
+    # Previous behavior was: 1) initialized attributes were the
```

Comment:
I think what it's trying to say is that the attributes hash was *always* the value of the attributes passed in, regardless of what `attr_*` methods were defined.  This change now lets you define the attributes so that 1) unset attributes are now nil and 2) you can change the attributes after initialization by using the attr_writer, rather than hackily mutating the return value of the `attributes` method 3) the return value of the `attributes` method is now frozen. since it is always derived, trying to mutate it is strongly discouraged :)